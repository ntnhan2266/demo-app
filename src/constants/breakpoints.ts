export const BREAKPOINTS = {
  small: 640,
  medium: 768,
  large: 1024,
  xlarge: 1280,
};
