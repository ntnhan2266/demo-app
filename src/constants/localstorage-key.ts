export const LOCALSTORAGE_KEY = {
  USERS: 'users'
}