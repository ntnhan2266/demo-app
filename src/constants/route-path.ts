export const ROUTE_PATH = {
  LOGIN: '/login',
  REGISTER: '/register',
  DASHBOARD: '/',
};
