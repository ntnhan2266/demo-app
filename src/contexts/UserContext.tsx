import React, { createContext, useState, ReactNode } from 'react';
import { PublicUser } from '@/interfaces/user';

// Context interface
interface IUserContext {
  user: PublicUser | null;
  setUser: (_user: PublicUser | null) => void;
}

// Create the context
export const UserContext = createContext<IUserContext | undefined>(undefined);

// Provider component
export const UserProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
  const [user, setUser] = useState<PublicUser | null>(null);

  const contextValue: IUserContext = {
    user,
    setUser,
  };

  return <UserContext.Provider value={contextValue}>{children}</UserContext.Provider>;
};
