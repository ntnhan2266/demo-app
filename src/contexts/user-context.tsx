import React, { createContext, useContext, useState, ReactNode } from 'react';
import { PublicUser } from '@/interfaces/user';

// Context interface
interface IUserContext {
  user: PublicUser | null;
  setUser: (user: PublicUser | null) => void;
}

// Create the context
const UserContext = createContext<IUserContext | undefined>(undefined);

// Provider component
export const UserProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
  const [user, setUser] = useState<PublicUser | null>(null);

  const contextValue: IUserContext = {
    user,
    setUser,
  };

  return <UserContext.Provider value={contextValue}>{children}</UserContext.Provider>;
};

// Custom hook to use the user context
export const useUser = () => {
  const context = useContext(UserContext);

  if (!context) {
    throw new Error('useUser must be used within a UserProvider');
  }

  return context;
};
