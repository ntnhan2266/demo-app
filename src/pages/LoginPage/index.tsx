import React from 'react';
import { Link } from 'react-router-dom';
import LoginForm from '@/components/LoginForm';
import { ROUTE_PATH } from '@/constants/route-path';

const RegisterPage: React.FC = () => {
  return (
    <div className='flex justify-center'>
      <div className='flex-1'>
        <div className='mb-4'>
          <LoginForm />
        </div>
        <div>
          <p className='text-center'>
            Don&apos;t have an account yet? &nbsp;
            <Link className='underline' to={ROUTE_PATH.LOGIN}>
              Register now.
            </Link>
          </p>
        </div>
      </div>
    </div>
  );
};

export default RegisterPage;
