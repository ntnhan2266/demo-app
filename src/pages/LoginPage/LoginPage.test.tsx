import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import LoginPage from '@/pages/LoginPage';
import { ROUTE_PATH } from '@/constants/route-path';
import { IUserContext, UserContext } from '@/contexts/UserContext';
import renderer from 'react-test-renderer';

// Mock the withAuthentication HOC
jest.mock('@/hocs/withAuthentication', () => (Component: React.FC) => Component);

describe('LoginPage', () => {
  const contextValue: IUserContext = {
    user: null,
    setUser: () => {},
  };

  test('should match the snapshot', () => {
    const tree = renderer
      .create(
        <UserContext.Provider value={contextValue}>
          <MemoryRouter>
            <LoginPage />
          </MemoryRouter>
        </UserContext.Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders login form', () => {
    render(
      <UserContext.Provider value={contextValue}>
        <MemoryRouter>
          <LoginPage />
        </MemoryRouter>
      </UserContext.Provider>,
    );

    // Ensure that the login form is rendered
    const loginFormElement = screen.getByTestId('login-form');
    expect(loginFormElement).toBeInTheDocument();
  });

  test('renders link to register page', () => {
    render(
      <UserContext.Provider value={contextValue}>
        <MemoryRouter>
          <LoginPage />
        </MemoryRouter>
      </UserContext.Provider>,
    );

    // Ensure that the link to register page is rendered
    const registerLinkElement = screen.getByText(/Register now/i);
    expect(registerLinkElement).toBeInTheDocument();
    expect(registerLinkElement).toHaveAttribute('href', ROUTE_PATH.REGISTER);
  });
});
