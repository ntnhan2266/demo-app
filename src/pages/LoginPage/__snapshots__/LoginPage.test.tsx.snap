// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should match the snapshot 1`] = `
<div
  className="container mx-auto"
  data-testid="login-page"
>
  <div
    className="mb-4 max-w-md mx-auto mt-8 p-6 bg-white rounded-md shadow"
  >
    <div
      className="mb-6"
    >
      <a
        className="text-black font-bold font-comfortaa text-4xl"
        href="/"
        onClick={[Function]}
      >
        βeta
      </a>
    </div>
    <div
      data-testid="login-form"
    >
      <h2
        className="text-2xl font-bold mb-4 uppercase"
      >
        Login to your account
      </h2>
      <form
        onSubmit={[Function]}
      >
        <div
          className="mb-4"
        >
          <label
            className="block w-full mb-1 text-xs font-medium text-gray-500 transition-all duration-200 ease-in-out "
            htmlFor="email"
          >
            Email or phone
            <sup
              className="text-red-600"
            >
              *
            </sup>
          </label>
          <input
            className="border rounded-md px-3 py-2 focus:outline-none transition duration-300 w-full md:w-96 max-w-full w-full md:w-full border-gray-400"
            data-testid="email-or-phone-input"
            disabled={false}
            id="email"
            name="emailOrPhone"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Enter your email or phone number"
            type="text"
          />
        </div>
        <div
          className="mb-4"
        >
          <label
            className="block w-full mb-1 text-xs font-medium text-gray-500 transition-all duration-200 ease-in-out "
            htmlFor="password"
          >
            Password
            <sup
              className="text-red-600"
            >
              *
            </sup>
          </label>
          <input
            className="border rounded-md px-3 py-2 focus:outline-none transition duration-300 w-full md:w-96 max-w-full w-full md:w-full border-gray-400"
            data-testid="password-input"
            disabled={false}
            id="password"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Enter your password"
            type="password"
          />
        </div>
        <button
          className="rounded-md focus:outline-none w-full text-white text-base px-4 py-2 bg-indigo-500 hover:bg-indigo-600"
          data-testid="submit-button"
          disabled={false}
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </div>
  <div>
    <p
      className="text-center"
    >
      Don't have an account yet?  
      <a
        className="underline"
        href="/register"
        onClick={[Function]}
      >
        Register now.
      </a>
    </p>
  </div>
</div>
`;
